Enumerable.rb

```diff
@@ -1,114 +1,131 @@
 module Enumerable
-    def my_each
-        return to_enum unless block_given?
-        length.times do |i|
-            yield(self[i])
-        end
-    end
+  def my_each
+    return to_enum unless block_given?
 
-    def my_each_with_index
-        return to_enum unless block_given?
-        length.times do |i|
-            yield(self[i], i)
-        end
+    length.times do |i|
+      yield(self[i])
     end
+  end
 
-    def my_select
-        return to_enum unless block_given?
-        arr = []
-        length.times do |i|
-            arr.push(self[i]) if yield(self[i])
-        end
-        arr
+  def my_each_with_index
+    return to_enum unless block_given?
+
+    length.times do |i|
+      yield(self[i], i)
     end
+  end
 
-    def my_all?
-        result = true
-        if block_given?
-            my_each { |i| result = false unless yield(i) }
-        else
-            my_each { |i| result = false unless i }
-        end 
-        result
+  def my_select
+    return to_enum unless block_given?
+
+    arr = []
+    length.times do |i|
+      arr.push(self[i]) if yield(self[i])
     end
+    arr
+  end
 
-    def my_any?(arg = nil)
-        result = false
-        if arg
-            my_each { |i| result = true if i.is_a?(arg) }
+  # rubocop:disable Metrics/CyclomaticComplexity
+  # rubocop:disable Metrics/PerceivedComplexity
+  def my_all?(arg = nil)
+    result = true
+    my_each do |i|
+      if result
+        if arg.is_a?(Class) || arg.is_a?(Regexp)
+          result = i.is_a?(arg) unless arg.is_a?(Regexp)
+          result = !i.match(arg).nil? if arg.is_a?(Regexp)
+        elsif arg
+          result = i == arg1
```

Comment:
- [ ] There's a typo in the line 38, inside my_all?
It should be just arg and not arg1 right?